### DISCUSSION TITLE
Why there is no env.local after build

### DISCUSSION THREAD

**Discussion by gzhhong** [vercel/next.js#48764] [Help] [answered]
### Summary

Hello, 

I put the environment variables in the .env.local and I suppose when I running the application in dev mode by `npm run dev`, the values in env.local can be referred in code by process.env.key_of_variable. But after I build the application by `npm run build`, I can see a .next folder is created and the pages and middleware are located in .next/server, but I can't find the .env.local file in the .next folder. After I deploy the application to cloud pod, I checked the files on the pod and I can't find the .env.local too. But the wired thing is the application can work well. I am a little bit confused about how the nextjs will use the .env.local after build. Anyone can give help?

Thanks,

James

**Comment by icyJoseph** [comment]
Hi,

The thing is that env vars that are used client side, `NEXT_PUBLIC_` are inlined, that is, copied into the client side bundle, so that you don't need to keep on including them when you deploy.

For server side variables, you need to keep on including them though:

- https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

> The value will be inlined into JavaScript sent to the browser because of the NEXT_PUBLIC_ prefix. This inlining occurs at build time, so your various NEXT_PUBLIC_ envs need to be set when the project is built.

**Reply by gzhhong** [reply]
Hello @icyJoseph Joseph, thanks for the information. So if I have some env variables must used on server side, I need copy the .env.local to the root folder of the application on server pod, right?

**Reply by icyJoseph** [reply, marked as answer]
Almost, but really, you just need to make sure they exist in the environment. One way to do that, is with `.env` files, yes.

**Reply by gzhhong** [reply]
Thanks @icyJoseph Joseph.
